### DISCUSSION TITLE
Supported Go Versions

### DISCUSSION THREAD

**Discussion by paganotoni** [gobuffalo/cli#185] [Ideas]
As we want to provide the best Developer experience possible with the Buffalo CLI and libraries we should focus on working on that experience instead of other maintenance tasks. At this moment we're maintaining multiple Go versions, which in some cases limits the APIs we can use for the tools.

There has been a not-so-public rule about supporting the last 2 versions of Go but in reality, we still run tests against 3 or 4 versions of Go. And some of our libraries are written with v1.13 API. As the Go team only actively supports two versions I think it would be good to embrace the same initiative and with that reduce the amount of work/considerations while working on Buffalo.

At this moment those 2 versions would be `v1.17` and `v1.18`. This would mean that while Buffalo may still work with Go v1.16 we will not keep that version into consideration while developing (nor our test suites). Eventually, Buffalo will stop working with Go 1.16 due to APIs we use.

I envision we could make this evident by:

- Maintaining a supported-versions document (maybe on the `.github` repo)
- Having the CLI warn about the outdated version. p.e. 
`[Warning] you're using an old version of Go (v1.16) for a better experience update to the latest v1.18)`

cc @sio4 @stanislas-m @fasmat

**Comment by sio4** [comment]
I agree with you, especially the following sentence. That is what I wanted to say in the previous discussion about version support. I think we should not be captured by consideration for the old versions while we don't need to proactively/intentionally break things.

> This would mean that while Buffalo may still work with Go v1.16 we will not keep that version into consideration while developing (nor our test suites).

I will soon update the test cases not to test against 1.16 anymore. (already done for stand-alone tests such as for `pop` and `fizz`)

One concern that came to me while I was considering updating `go.mod` file is that the `go.mod` format for 1.17 or higher is not easy to find an intentional version fix with an indirect comment directive. Any idea?


See also: https://github.com/gobuffalo/buffalo/issues/2264

**Reply by sio4** [reply]
just updated the standard test case which is used by all packages except `pop` and `fizz`. so from now tests will run for only 1.17 and 1.18.
